src/main/java/slimeknights/tconstruct/library/recipe/tinkerstation/building/ToolBuildingRecipe.java

```diff
@@ -48,13 +52,44 @@ public class ToolBuildingRecipe implements ITinkerStationRecipe {
   @Getter
   protected final IModifiable output;
   protected final int outputCount;
+  @Nullable
+  protected ResourceLocation layoutSlot;
   protected final List<Ingredient> ingredients;
 
+  /** @deprecated Use {@link #ToolBuildingRecipe(ResourceLocation, String, IModifiable, int, ResourceLocation, List)} */
+  @Deprecated
+  public ToolBuildingRecipe(ResourceLocation id, String group, IModifiable output, int outputCount, List<Ingredient> ingredients) {
+    this(id, group, output, outputCount, null, ingredients);
+  }
+
+  public ToolBuildingRecipe(ResourceLocation id, String group, IModifiable output, int outputCount, ResourceLocation layoutSlot, List<Ingredient> ingredients) {
+    this.id = id;
+    this.group = group;
+    this.output = output;
+    this.outputCount = outputCount;
+    this.layoutSlot = layoutSlot;
+    this.ingredients = ingredients;
+  }
+
+  public ResourceLocation getLayoutSlotId() {
+    return Objects.requireNonNullElse(layoutSlot, getOutput().getToolDefinition().getId());
+  }
+
+  public List<IToolPart> getToolParts() { return ToolPartsHook.parts(getOutput().getToolDefinition()); }
+
   /** Gets the additional recipe requirements beyond the tool parts */
   public List<Ingredient> getExtraRequirements() {
     return ingredients;
   }
 
+  public List<LayoutSlot> getLayoutSlots() {
+    return StationSlotLayoutLoader.getInstance().get(getLayoutSlotId()).getInputSlots();
+  }
+
+  public boolean isBroadTool() {
+    return getToolParts().size() >= 4;
```

Comment:
You also need to check `getExtraRequirements()`'s size, its the sum of the two being 4 or more that is "broad tools" (though I think a name like `requiresAnvil` might be better)